arch/src/aarch64/fdt.rs

```diff
@@ -0,0 +1,610 @@
+// Copyright 2020 Arm Limited (or its affiliates). All rights reserved.
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Portions Copyright 2017 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the THIRD-PARTY file.
+
```

Comment:
Done. I tried to adding one files per commit, however if we need to build each commit, we need to also modify the common code (for example those in `mod.rs` and `lib.rs`). Therefore I tried to import each module (memory layout, GIC, registers, FDT) per commit instead of each file per commit instead. Hope this is also ok to you :)

Reply:
Thanks it's totally fine! And actually per module is obviously better :)